test/test.js

```diff
@@ -14,6 +14,19 @@ test('missing file', (assert) => {
     .on('error', (err) => {
       assert.ok(err, 'errored');
       assert.ok(err.message.indexOf('ENOENT') === 0, 'expected error');
+      assert.equals(err.code, 'EINVALID', 'expected error code');
```

Comment:
👍  Agreed. Just committed the change. Now passing the raw error if an unexpected error occurs during `getFileType()`. `mapbox-file-sniff` will add `EINVALID`, otherwise dataset-unpacker just passes the error as-is.